src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.Converters.cs

```diff
@@ -20,48 +20,54 @@ public sealed partial class JsonSerializerOptions
         private static readonly Dictionary<Type, JsonConverter> s_defaultSimpleConverters = GetDefaultSimpleConverters();
 
         // The global list of built-in converters that override CanConvert().
-        private static readonly List<JsonConverter> s_defaultFactoryConverters = GetDefaultConverters();
+        private static readonly JsonConverter[] s_defaultFactoryConverters = new JsonConverter[]
+        {
+            // Nullable converter should always be first since it forwards to any nullable type.
+            new NullableConverterFactory(),
+            new EnumConverterFactory(),
+            new KeyValuePairConverterFactory(),
+            new IEnumerableConverterFactory(),
```

Comment:
Please add comments back here - IEnumerableConverterFactory should always be second to last and ObjectConverterFactory should always be last.